orderer/common/multichannel/registrar.go

```diff
@@ -498,3 +589,11 @@ func (r *Registrar) RemoveChannel(channelID string, removeStorage bool) error {
 	//TODO
 	return errors.New("Not implemented yet")
 }
+
+func (r *Registrar) removeJoinBlock(channelID string) error {
+	if err := r.joinBlockFileRepo.Remove(channelID); err != nil {
+		return errors.Wrapf(err, "failed removing joinblock for channel %s", channelID)
+	}
+
+	return nil
+}
```

Comment:
We should have a version of this exported and protected with a mutex, so that when a follower finishes on boarding it will be able to call and remove the join block.

Reply:
Is there a particular reason the follower should handle removing the join block and not the registrar itself? For now, I've added the remove logic into SwitchFollowerToChain.

Reply:
When the chain progresses past the join block, there is no reason to keep it around... there is no harm in that, but it still feels a bit weird.